CNV-14966: Virtualization Overview

lmandavi opened the pull request:
This PR is associated with [CNV-14966](https://issues.redhat.com/browse/CNV-14966) and it goes out in the CNV 4.10 release.

lmandavi commented:
https://deploy-preview-39721--osdocs.netlify.app/openshift-enterprise/latest/virt/logging_events_monitoring/virt-reviewing-virtualization-overview.html

lmandavi commented:
@yaacov, @fkargbo@redhat.com - Please review the content and provide your input.

yaacov commented:
cc:// @pcbailey

fkargbo commented:
@lmandavi Looks good to me!

fkargbo commented:
@lmandavi A few things:

1. From what I understand, the overview page will be limited only to cluster admins for the 4.10 release
2. The reference to the 'monitoring role' might not apply yet to this release as only cluster admins will be able to view this information for now
3. The Top Consumer card will be released with 7 monitored resources for the 4.10 release and more will beadded in the future.

(*Update) The current 7 monitored resources are:
- CPU
- Memory 
- Used filesystem
- Memory swap
- vCPU wait
- Storage throughput
- IOPS

cc:// @yaacov @ronensdeor

yaacov commented:
> @lmandavi A few things:
> 
> 1. From what I understand, the overview page will be limited only to cluster admins for the 4.10 release
> 2. The reference to the 'monitoring role' might not apply yet to this release as only cluster admins will be able to view this information for now

Yes, we will only be ready with the 'admin-role' page for 4.10
the 'monitoring-role' and 'project-admin-role' will not be in 4.10

p.s.
we will try to push the 'project-admin-role' and  'monitoring role' views into 4.10, but it will be best not to document it.

lmandavi commented:
@fkargbo @yaacov - Noting the changes you mention. Plan to update the doc accordingly.

pcbailey commented on modules/virt-about-monitored-resources-openshift-virtualization.adoc:
```diff
@@ -0,0 +1,26 @@
+|Details
+|A brief overview of service and version information for *{VirtProductName}*.
+|Status
+|Alerts for virtualization, storage, and networking.
```
Storage isn't part of the MVP for 4.10. Currently, only virtualization and networking are part of the Status card.

lmandavi replied:
Removed storage.

pcbailey commented on modules/virt-about-top-consumers-projects-vms-nodes.adoc:
```diff
@@ -0,0 +1,32 @@
+[id="virt-about-reviewing-top-consumers-projects-vms-nodes_{context}"]
+= About top consumers for projects, virtual machines, and nodes
+
+The *Top Consumers* cards display projects, virtual machines or nodes with maximum consumption of a resource. You can select a project, a virtual machine, or a node and view the top five consumers of a specific resource.
```
The user can select whether they would like to see the top 5 or top 10 consumers.

lmandavi replied:
Updated the content.

pcbailey commented on modules/virt-about-top-consumers-projects-vms-nodes.adoc:
```diff
@@ -0,0 +1,32 @@
+
+[NOTE]
+====
+Viewing the maximum resource consumption is limited to the top five consumers.
```
Same as above. User can select to see the top 5 or top 10 consumers.

lmandavi replied:
Updated the content.

pcbailey commented on modules/virt-about-top-consumers-projects-vms-nodes.adoc:
```diff
@@ -0,0 +1,32 @@
+|CPU
+|Projects, virtual machines, or nodes consuming the most CPU.
+|Memory
+|Projects, virtual machines, or nodes consuming the most memory (in bytes).
```
I'm not sure if this matters, but the unit of the value adjusts automatically depending on how large the value is, so while the base unit is in bytes, a value may be displayed in MiB, GiB, etc. The same applies to the used filesystem metric as well.

lmandavi replied:
I've added a brief explanation here, based on your input.

pcbailey commented on modules/virt-about-top-consumers-projects-vms-nodes.adoc:
```diff
@@ -0,0 +1,32 @@
+|vCPU wait
+|Projects, virtual machines, or nodes experiencing the maximum wait time (in seconds) for the vCPUs.
+|Storage throughput
+|Projects, virtual machines, or nodes with the highest data transfer rate to and from the storage media (in mbps).
```
This value adjusts automatically as well. It could be MBps or GBps depending on the value.

lmandavi replied:
I've added a brief explanation here, based on your input.

pcbailey commented on modules/virt-about-virtualization-overview.adoc:
```diff
@@ -0,0 +1,35 @@
+[id="virt-about-reviewing-top-consumers_{context}"]
+= About reviewing top consumers
+
+In the {VirtProductName} dashboard, you can select a specific time period and view the top consumers of resources within that time period. Top consumers are virtual machines or `virt-launcher` pods that are consuming the highest amount of resources.
```
This is about the CNV Dashboard and not the Virtualization Overview, correct?

lmandavi replied:
This module has been removed - please ignore.

pcbailey commented on modules/virt-reviewing-top-consumers-projects-vms-nodes.adoc:
```diff
@@ -0,0 +1,22 @@
+
+. From a *Top Consumer* card, select the type of resource from the drop-down menu: *CPU*, *Memory*, *Used Filesystem*, *Memory Swap*, *vCPU Wait*, or *Storage Throughput*.
+
+. Select *By Project*, *By VM*, or *By Node*. The top five consumers with the highest resource consumption display.
```
There is a dropdown that allows the user to select whether they want to see the top 5 or top 10 consumers.

lmandavi replied:
Updated the content.

lmandavi commented:
@pcbailey - Please review the updated content and provide your input:
https://deploy-preview-39721--osdocs.netlify.app/openshift-enterprise/latest/virt/logging_events_monitoring/virt-reviewing-virtualization-overview.html

fkargbo commented:
@lmandavi Looks good to me! A few things I noticed are: 
- No mention of the 'Permissions card' 
- The link to the Virtualization Dashboard (seen on the Top Consumers card) 
- Only the Top Consumers card is limited to the cluster-admin role (@yaacov can we confirm this?)
Thanks!

yaacov commented:
> Only the Top Consumers card is limited to the cluster-admin role (@yaacov can we confirm this?)

a. The "cluster overview page" should not be available for non admins.
b. we are missing the "project overview page", it is considered a bug, and we aim to fix it.
b. I don't know how the "project overview page" looks like, @fkargbo I was on PTO so I may be missing updates, do we have a design for it ?

fkargbo commented:
@yaacov regarding the 'project overview page' I'm not aware of the specifics for it.

lmandavi commented:
@fkargbo, @yaacov - Since I am not seeing "project overview page" in the cluster build provided by @gouyang and this feature is being worked on as a bug, I recommend that content development for "project overview page" be performed as a separate effort for the bug so that this PR can move forward with the current feature set that has been documented.

lmandavi commented:
@fkargbo - Please see my responses below.

> @lmandavi Looks good to me! A few things I noticed are:
> 
> * No mention of the 'Permissions card'
Adding this info.

> * The link to the Virtualization Dashboard (seen on the Top Consumers card)
My thought is that the link to the Virtualization Dashboard is self-explanatory and easy to understand what that link does. Will evaluate if I can add this info to one of the modules. 

> * Only the Top Consumers card is limited to the cluster-admin role (@yaacov can we confirm this?)
Request @yaacov to confirm.

Can make the change from pie chart to donut.

ronensdeor commented:
@Lavanya Mandavilli ***@***.***>

On Tue, Jan 4, 2022 at 6:28 PM Lavanya Mandavilli ***@***.***>
wrote:

> @fkargbo <https://github.com/fkargbo> - Please see my responses below.
>
> @lmandavi <https://github.com/lmandavi> Looks good to me! A few things I
> noticed are:
>
>    - No mention of the 'Permissions card'
>    Adding this info.
>
>
>    - The link to the Virtualization Dashboard (seen on the Top Consumers
>    card)
>    My thought is that the link to the Virtualization Dashboard is
>    self-explanatory and easy to understand what that link does. Will evaluate
>    if I can add this info to one of the modules.
>
> As you said, we don't need to explain what is the link, but it's a way to
explain that there is a way to see all top consumers (using the CNV
monitoring dashboard) and not just the top 5 / 10. It might even be useful
to add a link to the CNV monitoring dashboard document.

>
>
>
>    - Only the Top Consumers card is limited to the cluster-admin role (
>    @yaacov <https://github.com/yaacov> can we confirm this?)
>    Request @yaacov <https://github.com/yaacov> to confirm.
>
> Can make the change from pie chart to donut.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/39721#issuecomment-1004955014>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQZTZAXK2DPABY7X6JWGJOLUUMN3BANCNFSM5JXQM2MA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

fkargbo commented:
@lmandavi Thanks for the update!

yaacov commented:
LGTM

 > I recommend that content development for "project overview page" be performed as a separate effort

@lmandavi @fkargbo @pcbailey hi, makes sense to me too,

The prerequisite to login as admin is required now, in the refuter we will need to update, when we have the project-overview that will be available to project-admins (non cluster-admins)

lmandavi commented:
@ronensdeor , @fkargbo , @yaacov - Requested some details for the Permissions Card on Gchat.

fkargbo commented:
Hi @lmandavi ,

I’ll defer question 1 to the UI team (@yaacov @pcbailey) and or the @ronensdeor for more clarity.

**Question 2**
Yes, this card informs the user what task they can and cannot perform. 
**Question 3**
For the 4.10 release, the user will only be able to see which task they can/cannot perform. From what I understand, we will add some self-service text and 'Learn more' links here in future releases. @yaacov , @ronensdeor , feel free to add more if I’m missing something here.
**Question 4**
From a high-level point of view, we want to let the user know of their read/write capabilities with regards to the virtualization cluster. Thanks!

lmandavi commented:
For this release, the basic description provided is sufficient. When more features are added to the Permissions Card, we can expand the description.

Moving this PR to QE review.

fkargbo commented:
Sounds good 👍

gouyang commented:
LGTM

lmandavi commented:
Requesting peer review.

sjhala-ccs commented on modules/virt-about-virtualization-overview.adoc:
```diff
@@ -0,0 +1,35 @@
+//
+
+[id="virt-about-reviewing-top-consumers_{context}"]
+= About reviewing top consumers
```
Again, this is more of a reference module and should likely be renamed.

lmandavi replied:
Please see my response in a previous comment section.

lmandavi replied:
Agree. Changed it to a reference module.

lmandavi replied:
This module is no longer used. It was replaced by a module with a different title.

sjhala-ccs commented on modules/virt-about-virtualization-overview.adoc:
```diff
@@ -0,0 +1,35 @@
+[id="virt-about-reviewing-top-consumers_{context}"]
+= About reviewing top consumers
+
+In the {VirtProductName} dashboard, you can select a specific time period and view the top consumers of resources within that time period. Top consumers are virtual machines or `virt-launcher` pods that are consuming the highest amount of resources.
```
s/are consuming/consume

lmandavi replied:
This module was replaced by a newer module that has been reviewed. Removing this module.

lmandavi replied:
This module does not belong to this PR. I had planned to use it in this PR, but decided not to. It belongs to the Dashboard assembly. When I deleted, Travis was not happy. So this module has been added back, but not being used in this assembly.

sjhala-ccs reviewed (commented):
Some nits and suggestions for you to consider, thanks Lavanya.

lmandavi commented on _topic_maps/_topic_map.yml:
```diff
@@ -3095,6 +3095,8 @@ Topics:
 - Name: Logging, events, and monitoring
   Dir: logging_events_monitoring
   Topics:
+  - Name: Reviewing virtualization overview
```
Hmmm - The CNV Dashboard is a separate feature in our product. The Virt Overview page is not being referred to as a dashboard in the UX design and implementation. Dashboard and this Virt Overview are being treated as independent features, and there is actually a link from the Virt Overview page to the CNV Dashboard. Most users are likely to use both features - the dashboard for very detailed CNV metrics and visual charts that are based on Prometheus queries. And the Virt Overview gives a very high level overview of top consumers, permissions, deployed templates etc.

sjhala-ccs replied:
Okay. So dashboard is not the right word. But the title as is feels incomplete, would you consider changing it to "Reviewing Virtualization Overview page"?

lmandavi commented on modules/virt-reviewing-top-consumers-projects-vms-nodes.adoc:
```diff
@@ -0,0 +1,22 @@
+
+. Navigate to the *Top Consumers* cards.
+
+. From a *Top Consumer* card, select the type of resource from the drop-down menu: *CPU*, *Memory*, *Used Filesystem*, *Memory Swap*, *vCPU Wait*, or *Storage Throughput*.
```
Updated the content.

lmandavi commented on modules/virt-reviewing-top-consumers-projects-vms-nodes.adoc:
```diff
@@ -0,0 +1,22 @@
+
+. From a *Top Consumer* card, select the type of resource from the drop-down menu: *CPU*, *Memory*, *Used Filesystem*, *Memory Swap*, *vCPU Wait*, or *Storage Throughput*.
+
+. Select *By Project*, *By VM*, or *By Node*. The top five or top ten consumers with the highest resource consumption display.
```
Updated the content. Inserted a step in the procedure to select top five or top ten consumers. That should make it easy for the user to understand that selecting from the menu determines whether five or ten consumers are displayed.

lmandavi commented on modules/virt-about-top-consumers-projects-vms-nodes.adoc:
```diff
@@ -0,0 +1,34 @@
+// * virt/logging_events_monitoring/virt-reviewing-virtualization-overview.adoc
+
+[id="virt-about-reviewing-top-consumers-projects-vms-nodes_{context}"]
+= About top consumers for projects, virtual machines, and nodes
```
I am hesitant to change the title for the module. My general thought is that reference modules offer optional type of content. This concept module contains information that is useful to the user BEFORE they perform a procedure (they learn about each resource and can determine which resource they want to view when they perform the procedure). I see this module as a precursor to the procedure module that comes right after it. Perhaps I could reword the intro to the table as follows: The follow table provides a description of resources monitored for top consumers. Let me know your thoughts.

lmandavi replied:
Upon going back and reviewing this module, it seems to fit more into a reference pattern as you noted - especially because it doesn't precede any tasks or actions to be taken. Changing this to a reference module.

lmandavi commented on modules/virt-about-monitored-resources-openshift-virtualization.adoc:
```diff
@@ -0,0 +1,28 @@
+//
+
+[id="virt-about-monitored-resources-openshift-virtualization_{context}"]
+= About monitored resources in {VirtProductName} Overview
```
The name of the page as it appears at the very top is: "Virtualization Overview." Therefore, I've updated the content to refer to that page.

We don't want to use the word "dashboard" because "Virtualization Overview" and "Dashboards" are both separate features although they are somewhat related to each other. We want to maintain separation between the two pages to avoid any confusion. 

Regarding the renaming of the module - since this information introduces the user to each of the "Cards" in the "Virtualization Overview" page, my logic is to leave it as it is. Each card on that page provides information about a specific resource that is monitored (for example, a donut chart showing the types of templates that have been deployed to VMs).

Attaching screenshots of the two different features: "Dashboards" and "Virtualization Overview".

![Screenshot from 2022-01-12 11-38-06](https://user-images.githubusercontent.com/55500825/149197036-cbc6853f-37e2-47e5-bfc1-db517abe0e7f.png)

![Screenshot from 2022-01-12 13-05-13](https://user-images.githubusercontent.com/55500825/149196849-d55cc78e-df0c-4c38-8b12-cae566ef9499.png)

lmandavi commented on modules/virt-about-monitored-resources-openshift-virtualization.adoc:
```diff
@@ -0,0 +1,28 @@
+[id="virt-about-monitored-resources-openshift-virtualization_{context}"]
+= About monitored resources in {VirtProductName} Overview
+
+The following table shows actively monitored resources in the *{VirtProductName} Overview* page. This information is useful when you need obtain relevant data and intervene to resolve a problem.
```
It is "Virtualization Overview" and I've updated the content to reflect this page name.

About relabeling from "resources" to "fields", here are some thoughts.
Each of the resources is presented on a "Card" in the page. Maybe consider:
"Cards with monitored resources" - would that be a possible title in the table?
In my conversations with the UX designer, he did not refer to these items as fields. He referred to them as cards.

![Screenshot from 2022-01-12 13-09-49](https://user-images.githubusercontent.com/55500825/149197847-3bb07f7a-fe01-40e8-9f13-862afaec3ee7.png)

![Screenshot from 2022-01-12 13-09-42](https://user-images.githubusercontent.com/55500825/149197873-5920a185-6e56-4f02-86a9-bae1ba7c9c56.png)

sjhala-ccs replied:
@lmandavi I understand the terminology for the Top Consumers cards and am fine with referring to the CPU, Memory, etc. as resources. 
In my comment I was talking about the second screenshot shown here - Details, Activity, Permissions, Inventory, etc. These items are "fields" or "metrics" rather than resources and labeling them as resources might cause confusion because both tables in this assembly have similar titles and column headings.

lmandavi replied:
Thanks for your note. Since these items are a mix of fields and metrics, I will look into updating the table column heading.

lmandavi replied:
Please note - the .adoc file that will be updated is: virt-resources-monitored-actively-in-Virtualization-Overview-page.adoc.

lmandavi replied:
Renamed the table heading: Monitored resources, fields, and metrics.

lmandavi commented on modules/virt-about-virtualization-overview.adoc:
```diff
@@ -0,0 +1,35 @@
+//
+
+[id="virt-about-reviewing-top-consumers_{context}"]
+= About reviewing top consumers
```
This module was replaced by a newer module that has been reviewed. Removing this module.

lmandavi commented on modules/virt-about-top-consumers-projects-vms-nodes.adoc:
```diff
@@ -0,0 +1,34 @@
+[id="virt-about-reviewing-top-consumers-projects-vms-nodes_{context}"]
+= About top consumers for projects, virtual machines, and nodes
+
+The *Top Consumers* cards display projects, virtual machines or nodes with maximum consumption of a resource. You can select a project, a virtual machine, or a node and view the top five or top ten consumers of a specific resource.
```
I updated the content by mentioning "Virtualization Overview" page. Please review.

lmandavi commented on modules/virt-reviewing-top-consumers-projects-vms-nodes.adoc:
```diff
@@ -0,0 +1,22 @@
+
+.Procedure
+
+. In the *Administrator* perspective in the {VirtProductName} web console, navigate to *Virtualization -> *Overview*.
```
Added.

lmandavi commented on modules/virt-reviewing-top-consumers-projects-vms-nodes.adoc:
```diff
@@ -0,0 +1,22 @@
+[id="virt-reviewing-top-consumers-projects-vms-nodes_{context}"]
+= Reviewing top consumers for projects, virtual machines, and nodes
+
+In the *Administrator* perspective, you can view the {VirtProductName} page where top consumers of resources are displayed.
```
Updated the content.

lmandavi commented:
Please note - the virt-about-reviewing-top-consumers.adoc file belongs to another assembly (Dashboard). It was deleted while working on this PR. I've added this file back - however, it does not belong to this assembly. Therefore, it is not part of this review. That file is referenced by the Dashboard assembly, and when it was deleted, Travis generated a build error. To fix the build error and restore the file back, I added it back.

sjhala-ccs commented on modules/virt-resources-monitored-actively-in-Virtualization-Overview-page.adoc:
```diff
@@ -0,0 +1,28 @@
+//
+
+[id="virt-resources-monitored-actively-in-Virtualization-Overview-page_{context}"]
+= Resources monitored actively in the Virtualization Overview page
```
As I mentioned in my other comment, the title and column headings of this table are very similar to the other table in this assembly. This might cause confusion. I would suggest changing the table title to "Virtualization Overview page metrics".

lmandavi replied:
How about: Monitored resources, fields, and metrics. I chose this heading because "permissions" does not fall under metrics.

lmandavi replied:
There are plans to add additional monitored items to the Virtualization Overview page that may not fall under metrics.

sjhala-ccs commented on modules/virt-resources-monitored-actively-in-Virtualization-Overview-page.adoc:
```diff
@@ -0,0 +1,28 @@
+[id="virt-resources-monitored-actively-in-Virtualization-Overview-page_{context}"]
+= Resources monitored actively in the Virtualization Overview page
+
+The following table shows actively monitored resources in the *Virtualization Overview* page. This information is useful when you need obtain relevant data and intervene to resolve a problem.
```
Suggestion: "The following table shows metrics displayed on the **Virtualization Overview** page. You can use these metrics to obtain relevant data and to resolve any issues."

lmandavi replied:
Revamped as follows:
"The following table shows actively monitored resources, metrics, and fields in the *Virtualization Overview* page. This information is useful when you need obtain relevant data and intervene to resolve a problem."

Permissions does not fall under metrics.

sjhala-ccs reviewed (commented):
This looks great overall, just a few things to clarify before merging.

sjhala-ccs commented:
/cherrypick enterprise-4.10

apinnick commented on virt/logging_events_monitoring/virt-reviewing-virtualization-overview.adoc:
```diff
@@ -0,0 +1,30 @@
+[id="prerequisites_{context}"]
+== Prerequisites
+
+To use the *vCPU wait* metric in the *Top Consumers* card, the `schedstats=enable` kernel argument must be applied to the `MachineConfig` object. This kernel argument enables scheduler statistics used for debugging and performance tuning and adds a minor additional load to the scheduler. See the xref:../../post_installation_configuration/machine-configuration-tasks.adoc#nodes-nodes-kernel-argumen ...
```
If this prerequisite applies only to "vCPU wait", maybe it could be added as a note in virt-resources-monitored-for-top-consumption.adoc. Or it could be kept in the assembly and added as a note after that module if you want to retain the machine configuration link.

apinnick commented on virt/logging_events_monitoring/virt-reviewing-virtualization-overview.adoc:
```diff
@@ -0,0 +1,30 @@
+[id="virt-reviewing-virtualization-overview"]
+= Reviewing Virtualization Overview
```
Suggestion: "Monitoring resources by using the Virtualization Overview page" or dashboard.
I think that might make the use case clearer. Otherwise, I am not sure the user would understand why it would be a good idea to review the Virt Overview pae.

apinnick commented on modules/virt-resources-monitored-for-top-consumption.adoc:
```diff
@@ -0,0 +1,34 @@
+====
+
+The following table shows resources monitored for top consumers.
+
```
I just noticed that a lot of this information also appears in [About reviewing top consumers](https://docs.openshift.com/container-platform/4.10/virt/logging_events_monitoring/virt-reviewing-vm-dashboard.html#virt-about-reviewing-top-consumers_virt-reviewing-vm-dashboard). Has the OpenShift Virtualization dashboard been replaced by the Virtualization Overview page? If so, we might have to deprecate the dashboard.

apinnick commented on modules/virt-resources-monitored-for-top-consumption.adoc:
```diff
@@ -0,0 +1,34 @@
+====
+
+The following table shows resources monitored for top consumers.
+
```
I just noticed that a lot of this information also appears in [About reviewing top consumers](https://docs.openshift.com/container-platform/4.10/virt/logging_events_monitoring/virt-reviewing-vm-dashboard.html#virt-about-reviewing-top-consumers_virt-reviewing-vm-dashboard). Has the OpenShift Virtualization dashboard been replaced by the Virtualization Overview page? If so, we might have to deprecate the dashboard.

ronensdeor replied:
The overview page displays the top 5 / 10 consumers only at current time, the dashboard shows similar information but also allows you to see historical data (as a graph) and is not limited to the top 5 / 10 VMs.

apinnick replied:
@ronensdeor 
Good point. So the overview shows current info while the dashboard shows historical info. We may want to add a note about the difference at some point.
